src/components/RequestInviteForm.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { getAuthMemberData } from '../connectors';
+import { getRequestInviteOperation } from '../operations';
+import { postOperation } from '../actions';
+import YoutubeVideo from './YoutubeVideo';
+
+interface Props {
+  toUid: string;
+  toMid: string;
+  fullName: string;
+  videoUrl: string;
+}
+
+class RequestInviteForm extends Component<Props> {
+    constructor(props: InviteState) {
+      super(props);
+      this.state = {fullName: "", videoUrl: "", errorMessage: ""};
+    }
+
+    setFullName(event: any) : void {
+      this.setState({ fullName: event.target.value })
+    }
+
+    setVideoURL(event: any) : void {
+      this.setState({ videoUrl: event.target.value })
+    }
+
+    handleOnSubmit(event: any) : void {
+      event.stopPropagation();
+
+      if (this.state.fullName.length <= 0) {
+        this.setState({ errorMessage: "Please enter you name" });
+      }
+
+      else if (this.state.videoUrl.length <= 0) {
+        this.setState({ errorMessage: "Please enter a valid url" });
+      }
+
+      else {
+        let requestOp = getRequestInviteOperation(
+          this.props.toUid,
+          this.props.toMid,
+          this.props.authMemberData.id,
+          this.props.authMemberData.get('mid'),
+          this.state.fullName,
+          this.state.videoUrl
+        );
+        this.props.postOperation(requestOp);
+      }
+    }
+
+    clearErrorMessage(e: any) : void {
+      this.setState({ errorMessage: "" });
+    }
+
+    render() {
+      return (
+        <div>
+          <b>Invite new users</b>
+          <div>The more users join raha, the better! Type in a trusted friends gmail address to invite them.</div>
+          <div>
+            <input
+              onChange={ e => this.setFullName(e) }
+              placeholder="Your full name (e.g. John Doe)"
+              className="InviteInput ExtraWidth"
+            />           
+          </div>
+          <div>
+            <input
+              onChange={ e => this.setVideoURL(e) }
+              placeholder="VideoUrl"
+              className="InviteInput ExtraWidth"
+            />           
+          </div>
+          <button className="InviteButton"
+            onClick={ e=> this.handleOnSubmit(e) }>
+            Invite me!
+          </button>
+          <div className="InviteError">{ this.state.errorMessage }</div>
+          <div className="InviteVideoContainer">
+            {this.state.videoUrl && <YoutubeVideo youtubeUrl={this.state.videoUrl} />}
+          </div>
+        </div>
+      );
+    }
+}
+
+function mapStateToProps(state) {
+  return { authMemberData: getAuthMemberData(state) };
```

Comment:
Let's also grab `toMember: state.uidToMembers[ownProps.toUid]` info so that we can display message like ``You are requesting invite from ${this.props.toMember.get('full_name'), please only do this if you have made an invite video with them and you fully trust them. Though it can be changed later, by default they will be the person selected as your admin to assist with account recovery and you give them your vote for the Raha Board elections``

Reply:
As I was trying to implement this, I realized I didn't clarify something: should the user signing up already have been logged in and have an `memberId`? And do they then have to visit an existing member's page to get the `toMemberId`?

I feel like it would maybe be easier to have them just input the `toMemberId` into a form element here. I also assumed that we have not yet generated a `memberId` for this person, and need to do it in js from their full name.

If so, how do you construct the member id from a full name?

Reply:
For this diff using another form element to get `toMemberId` sounds good! Then later can think about it being a button on profile page or coming from a link.

We need to have the user sign up first so that they have an auth uid, else they cannot submit the REQUEST_INVITE op because of the [firestore rule](https://github.com/rahafoundation/raha.io/blob/d0d53bbda167b9d55fc195b5172b60a51b6587ca/firestore.rules#L22).

For sign up we have their full name (`state.auth.firebaseUser. displayName`) and then want to take that and lowercase, copy function from [this branch](https://github.com/rahafoundation/raha.io/blob/60625a9d48a1840f475a38ca1b5032bbc37137e1/functions/src/index.ts#L26) but now we are using `$` instead of `#` to separate name from public pin.

Reply:
added logic for getting `displayName` and constructing `creatorMid`